transit.md

```diff
@@ -167,16 +167,71 @@ The Transit client can attempt connections to multiple relays, and uses the
 first one that passes negotiation. Each side combines a locally-configured
 hostname/port (usually "baked in" to the application, and hosted by the
 application author) with additional hostname/port pairs that come from the
-peer. This way either side can suggest the relays to use. The `wormhole`
-application accepts a `--transit-helper tcp:myrelay.example.org:12345`
-command-line option to supply an additional relay. The connection hints
+peer. This way either side can suggest the relays to use. The connection hints
 provided by the Transit instance include the locally-configured relay, along
 with the dynamically-determined direct hints. Both should be delivered to the
 peer.
 
+## Canonical hint encodings
+
+The transit protocol relies on an existing secured (but possibly low-bandwidth)
+communication channel to exchange the abilities and hints. Thus, it has no
+influence over how they are encoded. However, we make a suggestion using JSON
+messages for other protocols to use.
+
+### `direct-tcp-v1`, `tor-tcp-v1`
+
+```json
+{
+  "type": "direct-tcp-v1" or "tor-tcp-v1",
+  "hostname": <string>,
+  "port": <u16>,
+  "priority": <float; optional>
+}
+```
+
+`hostname` must be compliant with the `host` part of an URL, i.e. it may be an
+IP address or a domain. Furthermore, IPv6 link-local addresses are not supported.
+
+### `relay-v1`
+
+```json
+{
+  "type": "relay-v1",
+  "hints": [
+    {
+      "hostname": "<string>",
+      "port": "<tcp port>",
+      "priority": "<number, usually [0..1], optional>"
+    },
+    …
+  ]
+}
+```
+
+Each entry in `hints` is equivalent to a `direct-tcp-v1` hint. Due to historic
+weirdness of the encoding, no more than one single relay server should be hinted at.
```

Comment:
That's the point: I actually don't know. The current specification has one `relay-v1` hint which contains a *list* of direct hints. But since I've never seen more than one hint (in total), I don't know what the semantics are. I'd have to dig up some code. My first guess was that there is only one `relay-v1` hint which contains all the relay servers, which would be different to what is done in `relay-v2`. But again, I don't know.